lib/snap_gene.ex

```diff
@@ -105,12 +105,23 @@ defmodule Bio.IO.SnapGene do
     sequence_module = Keyword.get(opts, :sequence_type, Bio.IO.SequenceTuple)
 
     case File.read(filename) do
-      {:ok, content} -> {:ok, struct(__MODULE__, parse(content, %{}, sequence_module))}
+      {:ok, content} -> {:ok, struct(__MODULE__, do_parse(content, %{}, sequence_module))}
       not_ok -> not_ok
     end
   end
 
-  defp parse(<<>>, output, _module), do: output
+  @doc"""
+  Identical to the read function except this takes a SnapGene file in binary form instead of a file
```

Comment:
I would re-word this to something like,

"Similar to `read/2` except that it takes a binary instead of a file path."

Then, in the typespec, you can use the special `%__MODULE__{}` to have the typespec define the return type correctly. No need to duplicate that documentation!